src/components/sbb-menu/sbb-menu.tsx

```diff
@@ -289,7 +289,9 @@ export class SbbMenu implements ComponentInterface {
       this._state = 'closed';
       this._dialog.firstElementChild.scrollTo(0, 0);
       // Manually focus last focused element in order to avoid showing outline in Safari
-      this._triggerElement?.focus();
+      this._triggerElement?.focus({
+        preventScroll: this._triggerElement.tagName === 'SBB-HEADER-ACTION',
```

Comment:
Agree with you and @jeripeierSBB  about the coupling problem.
I chose to add the logic inside the sbb-menu because the sbb-header-action doesn't have knowledge of if it is attached to any menu and probably the fix would have to be messier.

Anyway, I added the comment to clarify the reason for the fix :)